spec/models/manager_refresh/inventory_collection_default_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe ManagerRefresh::InventoryCollectionDefault do
+  before { allow(MiqServer).to receive(:my_zone) { 'My zone' } }
+  let(:zone) { Zone.first }
+
+  context 'Amazon' do
+    before do
+      ems = new_amazon_ems
+      with_vcr_data { expect { EmsRefresh.refresh(ems) }.to change { VmOrTemplate.count }.from(0) }
+      expect { ems.destroy }.to_not change { VmOrTemplate.count }
+      expect(ExtManagementSystem.count).to be_zero
+    end
+
+    it 'pickups existing Vm records on EMS re-adding' do
+      expect(Vm.count).to_not be_zero
```

Comment:
@AlexanderZagaynov unless I am mistaken, each of the specs will cause a refresh? That is why we don't do the normal clean way of writing specs (test 1 thing). But we run refresh once and just do all the assert sequentially. (refresh can take units or dozens of seconds, so that quickly adds up)

Reply:
done